src/openapi-generator.ts

```diff
@@ -84,12 +83,14 @@ export class OpenAPIGenerator {
     name: string;
     component: OpenAPIComponentObject;
   }[] = [];
+  private openAPIVersion: string | undefined;
```

Comment:
Hmmm a few things here.
1. How do you feel about making this an enum instead of a plain string. That way we would be the ones defining the possible/suported OpenAPI versions.
2. I would make this required and also a part of the `OpenAPIGenerator` constructor instead of using it upon the method invocation.
3. For `generateDocument` we can make `openapi` optional - if it is not provided we would use the value from the constructor. If it is provided - we would use it. That way the generation logic would be explicit and at the same time we would leave an option for the user to spcify the version at his own risk.

Reply:
afaik this is the main breaking change for open API 3.1.0. The other changes are listed in the article above. example -> examples consumers can change manually. I can have a look into the union/array nullables.